### PR TITLE
common: fix KW issues and other fixes

### PR DESCRIPTION
**Pull request by wlemkows**
This change is [Reviewable](https://reviewable.io/reviews/pmem/miniasync/56)

### CODE REVIEW

File: examples/basic-async/basic-async.c [hunk 1]
```diff
@@ -33,15 +34,30 @@ main(void)
 	 */
 	for (int i = 0; i < 2; i++) {
 		char *src1 = malloc(TEST_SIZE * sizeof(char));
```
Comment (pbalcer) [inline examples/basic-async/basic-async.c:36]: if this were a test I wouldn't complain, but in an example, I think this is just too bleh.
I think we can create a function from the inner parts of this loop, and use goto error handling to manage the src/dst buffers.

**Comment by kswiecicki** [comment]
___
*[src/core/os.h, line 106 at r1](https://reviewable.io/reviews/pmem/miniasync/56#-MwfJJVJ2BznuB3bD6dH:-MwfJJVK1esZBukgG3mg:b-hyoy0j) ([raw file](https://github.com/pmem/miniasync/blob/e23ab21b18be572b512df4683195f62d3b4ffb6a/src/core/os.h#L106)):*
> ```objc
> char *os_getenv(const char *name);
> const char *os_strsignal(int sig);
> int os_execv(const char *path, char *const argv[]);
> ```

Why was this removed?



_Code quote:_
```objc
int os_execv(const char *path, char *const argv[]);
```

**Review by wlemkows — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/56)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @kswiecicki and @pbalcer)

___
*[examples/basic-async/basic-async.c, line 36 at r1](https://reviewable.io/reviews/pmem/miniasync/56#-Mw_pya96VvbeN4GcjG2-r1-36:-Mx-MPWEEwIRWF88x9qj:b-896fix) ([raw file](https://github.com/pmem/miniasync/blob/b639c681f5485ec1db4215aa9ce307d04eb3781e/examples/basic-async/basic-async.c#L36)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

if this were a test I wouldn't complain, but in an example, I think this is just too bleh.
I think we can create a function from the inner parts of this loop, and use goto error handling to manage the src/dst buffers.
</blockquote></details>

Done.
___
*[src/core/os.h, line 106 at r1](https://reviewable.io/reviews/pmem/miniasync/56#-MwfJJVJ2BznuB3bD6dH:-Mx-Mx-M31tus39P0O-c:b-rvnre8) ([raw file](https://github.com/pmem/miniasync/blob/e23ab21b18be572b512df4683195f62d3b4ffb6a/src/core/os.h#L106)):*
<details><summary><i>Previously, kswiecicki (Krzysztof Święcicki) wrote…</i></summary><blockquote>

Why was this removed?
</blockquote></details>

Using this function causes errors in KW, it is not needed in the miniasync, we used it only for pmempool in PMDK.

**Review by DamianDuy — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/56#-:-Mx-YLMC41J6TYOOI4eC:bnfp4nl)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @kswiecicki and @pbalcer)

**Comment by kswiecicki** [comment]
:lgtm:

**Review by pbalcer — APPROVED** [review]
:lgtm:

> Reviewed 2 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/pmem/miniasync/56#-:-Mx-ZpBz1lbxhR4ebVJv:bnfp4nl)* status: 2 of 7 files reviewed, 1 unresolved discussion (waiting on @kswiecicki and @pbalcer)

**Pull request merged**